Framework/LiveData/src/ISIS/ISISKafkaEventStreamDecoder.cpp

```diff
@@ -285,6 +285,8 @@ void ISISKafkaEventStreamDecoder::captureImplExcept() {
   m_extractedEndRunData = true;
   std::string buffer;
   while (!m_interrupt) {
+    if (getBufferMemorySizeBytes() > MAX_BUFFER_SIZE)
+      continue;
```

Comment:
Could you try adding a thread sleep for 50ms here before the `continue` please?
It isn't currently giving the other thread opportunity to grab the mutex to empty the buffer.